hobbit-gui/gui-serverbackend/src/main/java/de/usu/research/hobbit/gui/rest/ExperimentsResources.java

```diff
@@ -65,28 +67,81 @@
+                                String systemURI = system.getURI();
+                                // check if it's owned by user
+                                if (userOwnedSystemIds.contains(systemURI)) {
+                                    results.add(RdfModelHelper.createExperimentBean(model, model.getResource(experimentUri)));
+                                }
+                            }
+                        }
                     }
                 }
             } else if (challengeTaskId != null) {
                 System.out.println("Querying experiment results for challenge task " + challengeTaskId);
-                // create experiment URI
-                String query = SparqlQueries.getExperimentOfTaskQuery(null, challengeTaskId, null);
-                // TODO make sure that the user is allowed to see the
-                // experiment!
+                // create experiment URI from public results graph
+                String query = SparqlQueries.getExperimentOfTaskQuery(null, challengeTaskId, Constants.PUBLIC_RESULT_GRAPH_URI);
+                // get public experiment
                 Model model = StorageServiceClientSingleton.getInstance().sendConstructQuery(query);
-                results = RdfModelHelper.createExperimentBeans(model);
+                // if model is public and available - go with it
+                if (model != null) {
+                    results = RdfModelHelper.createExperimentBeans(model);
+                } else {
+                    // otherwise try to look in private graph
+                    query = SparqlQueries.getExperimentOfTaskQuery(null, challengeTaskId, Constants.PRIVATE_RESULT_GRAPH_URI);
+                    model = StorageServiceClientSingleton.getInstance().sendConstructQuery(query);
+                    if (model != null) {
+                        // get challenge organizer
+                        Resource subj = model.getResource(challengeTaskId);
+                        // get challenge info
+                        String challengeQuery = SparqlQueries.getChallengeTaskOrganizer(challengeTaskId, null);
+                        Model challengeModel = StorageServiceClientSingleton.getInstance().sendConstructQuery(challengeQuery);
+                        Resource organizer = RdfHelper.getObjectResource(challengeModel, subj, HOBBIT.organizer);
+                        if (organizer != null) {
+                            // check if organizer is user
+                            // return whole thing if he is
+                            if (organizer.getURI() == userInfo.getPreferredUsername()) {
+                                results = RdfModelHelper.createExperimentBeans(model);
+                            } else {
+                                // if he is not, iterate over the beans and remove all beans that's now user owned
+                                List<ExperimentBean> experiments = RdfModelHelper.createExperimentBeans(model);
```

Comment:
This would be the second position at which the set of user systems is needed and needs to be loaded.